### PR TITLE
*: batch process peer task

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #7897.

### What is changed and how does it work?

```
$ benchstat old.txt new.txt
goos: linux
goarch: amd64
pkg: github.com/tikv/pd/server/cluster
cpu: Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz
                    │    old.txt    │               new.txt                │
                    │    sec/op     │    sec/op     vs base                │
HandleStatsAsync-72   610.2n ± 171%   370.1n ± 27%  -39.35% (p=0.000 n=10)

                    │   old.txt   │              new.txt               │
                    │    B/op     │    B/op     vs base                │
HandleStatsAsync-72   128.00 ± 3%   24.00 ± 0%  -81.25% (p=0.000 n=10)

                    │  old.txt   │              new.txt               │
                    │ allocs/op  │ allocs/op   vs base                │
HandleStatsAsync-72   5.000 ± 0%   3.000 ± 0%  -40.00% (p=0.000 n=10)
```

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/statistics/hot_peer_cache.go [hunk 1]
```diff
@@ -174,58 +174,65 @@ func (f *hotPeerCache) collectExpiredItems(region *core.RegionInfo) []*HotPeerSt
...
-	source := utils.Direct
-	if oldItem == nil {
-		for _, storeID := range f.getAllStoreIDs(region) {
-			oldItem = f.getOldHotPeerStat(regionID, storeID)
-			if oldItem != nil && oldItem.allowInherited {
-				source = utils.Inherit
-				break
+
+	regionPeers := region.GetPeers()
+	var stats []*HotPeerStat
```
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/statistics/hot_peer_cache.go:186]: the number of stats is not fixed

[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/statistics/hot_peer_cache.go]: done

File: pkg/statistics/hot_peer_cache.go [hunk 2]
```diff
@@ -174,58 +174,65 @@ func (f *hotPeerCache) collectExpiredItems(region *core.RegionInfo) []*HotPeerSt
...
+		for i, peer := range regionPeers {
+			newItem.stores[i] = peer.GetStoreId()
+		}
+		if oldItem == nil {
+			if stat := f.updateNewHotPeerStat(newItem, deltaLoads, time.Duration(interval)*time.Second); stat != nil {
+				stats = append(stats, stat)
+			}
+			continue
+		}
+		if stat := f.updateHotPeerStat(region, newItem, oldItem, deltaLoads, time.Duration(interval)*time.Second, source); stat != nil {
```
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/statistics/hot_peer_cache.go]: done

**Review by JmPotato — APPROVED** [review]

File: pkg/statistics/hot_cache.go [hunk 3]
```diff
@@ -172,14 +173,14 @@ func (w *HotCache) Update(item *HotPeerStat, kind utils.RWType) {
 
 // CheckWritePeerSync checks the write status, returns update items.
 // This is used for mockcluster, for test purpose.
-func (w *HotCache) CheckWritePeerSync(peer *core.PeerInfo, region *core.RegionInfo) *HotPeerStat {
-	return w.writeCache.checkPeerFlow(peer, region)
+func (w *HotCache) CheckWritePeerSync(region *core.RegionInfo, peers []*metapb.Peer, loads []float64, interval uint64) []*HotPeerStat {
```
Comment (lhy1024) [inline pkg/statistics/hot_cache.go:176]: `CheckWritePeersSync` or `CheckWriteRegionSync`?

File: pkg/statistics/hot_cache.go [hunk 4]
```diff
@@ -172,14 +173,14 @@ func (w *HotCache) Update(item *HotPeerStat, kind utils.RWType) {
...
+	return w.writeCache.checkPeerFlow(region, peers, loads, interval)
 }
 
 // CheckReadPeerSync checks the read status, returns update items.
 // This is used for mockcluster, for test purpose.
-func (w *HotCache) CheckReadPeerSync(peer *core.PeerInfo, region *core.RegionInfo) *HotPeerStat {
-	return w.readCache.checkPeerFlow(peer, region)
+func (w *HotCache) CheckReadPeerSync(region *core.RegionInfo, peers []*metapb.Peer, loads []float64, interval uint64) []*HotPeerStat {
```
Comment (lhy1024) [inline pkg/statistics/hot_cache.go:182]: ditto

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: pkg/statistics/hot_cache.go (hunk 3 above)
Comment (rleungx) [reply pkg/statistics/hot_cache.go:176]: IMO, peer is better

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**